scipy/stats/_correlation.py

```diff
@@ -0,0 +1,200 @@
+    when the association is not monotonic.
+
+    Parameters
+    ----------
+    x, y : array-like
+        The samples: corresponding observations of the independent and
+        dependent variable. The (N-d) arrays must be broadcastable.
+    axis : int, default: 0
+        Axis along which to perform the test.
+    method : 'asymptotic' or `PermutationMethod` instance, optional
+        Selects the method used to calculate the *p*-value.
+        Default is 'asymptotic'. The following options are available.
+
+        * ``'asymptotic'``: compares the standardized test statistic
+          against the normal distribution.
+        * `PermutationMethod` instance. In this case, the p-value
+          is computed using `permutation_test` with the provided
+          configuration options and other appropriate settings.
+
+    y_continuous : bool, default: False
+        Whether `y` is assumed to be drawn from a continuous distribution.
+        If `y` is drawn from a continuous distribution, results are valid
+        whether this is assumed or not, but enabling this assumption will
+        result in faster computation and typically produce similar results.
+
+    Returns
+    -------
+    res : SignificanceResult
+        An object containing attributes:
+
+        statistic : float
+            The xi correlation statistic.
+        pvalue : float
+            The associated *p*-value: the probability of a statistic at least as
+            high as the observed value under the null hypothesis of independence.
+
+    See Also
+    --------
+    scipy.stats.pearsonr, scipy.stats.spearmanr, scipy.stats.kendalltau
+
+    References
+    ----------
+    .. [1] Chatterjee, Sourav. "A new coefficient of correlation." Journal of
+           the American Statistical Association 116.536 (2021): 2009-2022.
+           :doi:`10.1080/01621459.2020.1758115`.
+
+    Examples
+    --------
+    Generate perfectly correlated data, and observe that the xi correlation is
+    nearly 1.0.
+
+    >>> import numpy as np
+    >>> from scipy import stats
+    >>> rng = np.random.default_rng(348932549825235)
+    >>> x = rng.uniform(0, 10, size=100)
+    >>> y = np.sin(x)
+    >>> res = stats.xi_correlation(x, y)
+    >>> res.statistic
+    np.float64(0.9012901290129013)
+
+    The probability of observing such a high value of the statistic under the
+    null hypothesis of independence is very low.
+
+    >>> res.pvalue
+    np.float64(2.2206974648177804e-46)
+
+    As noise is introduced, the correlation coefficient decreases.
+
+    >>> noise = rng.normal(scale=[[0.1], [0.5], [1]], size=(3, 100))
+    >>> res = stats.xi_correlation(x, y + noise, axis=-1)
+    >>> res.statistic
+    array([0.79507951, 0.41824182, 0.16651665])
+
+    Because the distribution of `y` is continuous, continuity may be
+    assumed with little effect on the results.
+
+    >>> stats.xi_correlation(x, y + noise, y_continuous=True, axis=-1).statistic
```

Comment:
Following my comment in the code, I think Y_continuous should affect the p-value and not the statistic value. (Here you obtained identical statistic values because the input data has no ties, so even with the current logic of the code the “if” branch and the “else” branch produce the same result.)

Reply:
Yup. I know. Let's close this comment though and discuss it above.